src/subgraph/abs.c

```diff
@@ -116,6 +116,38 @@ static enum xnn_status setup_abs_operator(
   }
 }
 
+static enum xnn_shape_inference_status infer_shape_forward(
+  const struct xnn_operator_data* opdata,
+  struct xnn_value* values,
+  size_t num_values)
+{
+  enum xnn_shape_inference_status status = xnn_shape_inference_status_no_change;
+
+  const uint32_t output_id = opdata->outputs[0];
+  struct xnn_value* output = &values[output_id];
+
+  // Propagate input shape to output.
+  const struct xnn_value* input = &values[opdata->inputs[0]];
+  for (size_t cur_dim = 0; cur_dim < input->shape.num_dims; cur_dim++) {
```

Comment:
```suggestion
  for (size_t cur_dim = 0; cur_dim < input->shape.num_dims - 1; cur_dim++) {
```
Nit: assert for channels not changing?